### PR TITLE
feat(stream_chat): upgrade to null safe dependencies

### PR DESCRIPTION
**Pull request by felangel**
# Submit a pull request

## CLA

- [X] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required).
- [X] The code changes follow best practices
- [X] Code changes are tested (add some information if not applicable)

## Description of the pull request
- feat(stream_chat): upgrade to null safe dependencies (part of #339)

### Plan

**Phase 1. Upgrade dependencies to null-safe compatible dependencies for each stream library**
  - [X] stream_chat
  - [ ] stream_chat_flutter
  - [ ] stream_chat_flutter_core
  - [ ] stream_chat_persistence

**Phase 2. Upgrade to sound null safety for each stream library**
  - [ ] stream_chat
  - [ ] stream_chat_flutter
  - [ ] stream_chat_flutter_core
  - [ ] stream_chat_persistence

### CODE REVIEW

**Comment by felangel** [comment]
The CI is failing for the other packages due to incompatible dependencies. I'm happy to migrate them all at once but I would normally do packages one at a time to make it easier to review and keep the PR focused.

Looks like `melos bootstrap` dynamically links them all so let me know what you want me to do, thanks! 👍

**Review by Nash0x7E2 — APPROVED** [review]
Great start, LGTM.

**Comment by Nash0x7E2** [comment]
@felangel Let's do one at a time, we'll work on the null safety branch until everything is migrated.

**Comment by Nash0x7E2** [comment]
@GetStream/flutter-developers can cherry-pick commits as needed.

**Review by imtoori — APPROVED** [review]
The example under `packages/stream_chat/example` is not running, nothing obvious, but probably an issue related to dio similar to this [one](https://github.com/flutterchina/dio/issues/1092)

Otherwise LGTM

**Comment by felangel** [comment]
@imtoori @Nash0x7E2 just pushed a fix and verified the example is working now, can you take another look? Thanks! 👍

File: packages/stream_chat_persistence/lib/src/dao/message_dao.dart [hunk 1]
```diff
@@ -135,8 +135,8 @@ class MessageDao extends DatabaseAccessor<MoorChatDatabase>
           _pinnedByUsers, messages.pinnedByUserId.equalsExp(_pinnedByUsers.id)),
     ])
           ..where(messages.channelCid.equals(cid))
-          ..where(
-              isNull(messages.parentId) | messages.showInChannel.equals(true))
+          ..where(messages.parentId.isNotNull() |
```
Comment (xsahil03x) [inline packages/stream_chat_persistence/lib/src/dao/message_dao.dart:138]: Consider changing this to `isNull`.

File: packages/stream_chat_persistence/lib/src/dao/pinned_message_dao.dart [hunk 2]
```diff
@@ -135,7 +135,7 @@ class PinnedMessageDao extends DatabaseAccessor<MoorChatDatabase>
           pinnedMessages.pinnedByUserId.equalsExp(_pinnedByUsers.id)),
     ])
           ..where(pinnedMessages.channelCid.equals(cid))
-          ..where(isNull(pinnedMessages.parentId) |
+          ..where(pinnedMessages.parentId.isNotNull() |
```
Comment (xsahil03x) [inline packages/stream_chat_persistence/lib/src/dao/pinned_message_dao.dart:138]: Consider changing this to `isNull`.

Comment (felangel) [reply packages/stream_chat_persistence/lib/src/dao/pinned_message_dao.dart]: Good catch! Need to stop coding at midnight 😛

**Pull request merged**